python/ray/serve/backend_worker.py

```diff
@@ -263,56 +235,28 @@ async def invoke_single(self, request_item: Query) -> Any:
         return result
 
     async def invoke_batch(self, request_item_list: List[Query]) -> List[Any]:
-        arg_list = []
-        kwargs_list = defaultdict(list)
-        context_flags = set()
-        batch_size = len(request_item_list)
+        args = []
         call_methods = set()
+        batch_size = len(request_item_list)
 
+        # Construct the batch of requests
         for item in request_item_list:
-            args, kwargs, is_web_context = parse_request_item(item)
-            context_flags.add(is_web_context)
-
-            call_method = self.get_runner_method(item)
-            call_methods.add(call_method)
-
-            if is_web_context:
-                # Python context only have kwargs
-                flask_request = args[0]
-                arg_list.append(flask_request)
-            else:
-                # Web context only have one positional argument
-                for k, v in kwargs.items():
-                    kwargs_list[k].append(v)
-
-                # Set the flask request as a list to conform
-                # with batching semantics: when in batching
-                # mode, each argument is turned into list.
-                if self.has_positional_args(call_method):
-                    arg_list.append(FakeFlaskRequest())
+            args.append(parse_request_item(item))
+            call_methods.add(self.get_runner_method(item))
 
         timing_start = time.time()
         try:
-            # Check mixing of query context (unified context needed).
-            if len(context_flags) != 1:
-                raise RayServeException(
-                    "Batched queries contain mixed context. Please only send "
-                    "the same type of requests in batching mode.")
-            serve_context.web = context_flags.pop()
-
             if len(call_methods) != 1:
                 raise RayServeException(
-                    "Queries contain mixed calling methods. Please only send "
-                    "the same type of requests in batching mode.")
-            call_method = ensure_async(call_methods.pop())
-
-            serve_context.batch_size = batch_size
-            # Flask requests are passed to __call__ as a list
-            arg_list = [arg_list]
+                    f"Queries contain mixed calling methods: {call_methods}. "
+                    "Please only send the same type of requests in batching "
+                    "mode.")
```

Comment:
This is pretty ugly. I guess we should have a queue per method type if we really want to support this